core/math/quick_hull.cpp

```diff
@@ -413,7 +414,12 @@ Error QuickHull::build(const Vector<Vector3> &p_points, Geometry::MeshData &r_me
 							Edge e2(idx, idxn);
 
 							Map<Edge, RetFaceConnect>::Element *F2 = ret_edges.find(e2);
-							ERR_CONTINUE(!F2);
+
+							if (unlikely(!F2)) {
+								warning_not_f2 = true;
```

Comment:
Could you use `WARN_PRINT_ONCE()` here instead of using booleans to defer it?

Reply:
I can easily change these to `WARN_PRINT_ONCE`s within the if statements.

This is just me, my reflex action is to write efficient code :grin: . Doing it like this involves one less if statement (and potentially branch), which can be important in hot loops. But it depends how the compiler optimizes it, and CPUs are very good at branch prediction in this kind of case so it probably doesn't make a lot of difference so I'll change it if it makes it more readable.

Reply:
Ah no on second thoughts, it would involve 3 if statements because of the `_flag_warnings` parameter. So it probably is best how it is, otherwise the logic in the hot loop will actually become harder to follow in terms of readability, as well as being potentially less efficient.

I.e. it would end up being:
```
if (unlikely(!F2))
{
    if (_flag_warnings)
    {
        WARN_PRINT_ONCE etc
    }
    continue;
}
```

Reply:
All that logic is quite awkward IMO.

Errors and warnings should point at things that are either:
- Engine bugs that we need to fix.
- User mistakes where they pass invalid input.

If it's neither of those and those are benign cases which are not meant to call the user's attention, they should just be replaced by silent `continue`. If they are signs that the user or the engine is using bogus data, that's something we should fix.